packages/quantic/force-app/main/default/lwc/quanticResultLink/__tests__/quanticResultLink.test.js

```diff
@@ -0,0 +1,113 @@
+// @ts-ignore
+import mockKnowledgeArticleResult from './data/knowledgeArticleResult.json';
+// @ts-ignore
+import mockSalesforceResult from './data/salesforceResult.json';
+
+jest.mock('c/quanticHeadlessLoader');
+
+function createTestComponent(options) {
+  const element = createElement('c-quantic-result-link', {
+    is: QuanticResultLink,
+  });
+
+  for (const [key, value] of Object.entries(options)) {
+    element[key] = value;
+  }
+
+  document.body.appendChild(element);
+  return element;
+}
+
+// Helper function to wait until the microtask queue is empty.
+function flushPromises() {
+  return new Promise((resolve) => setTimeout(resolve, 0));
+}
+
+// Helper function to mock headless for this suite of tests.
+function mockHeadless() {
+  jest.spyOn(mockHeadlessLoader, 'getHeadlessBundle').mockReturnValue({
+    buildInteractiveResult: () => ({
+      select: jest.fn(),
+      beginDelayedSelect: jest.fn(),
+      cancelPendingSelect: jest.fn(),
+    }),
+  });
+
+  jest.spyOn(mockHeadlessLoader, 'getHeadlessEnginePromise').mockReturnValue(
+    new Promise((resolve) => {
+      resolve();
+    })
+  );
+}
+
+describe('c-quantic-result-link', () => {
+  function cleanup() {
+    // The jsdom instance is shared across test cases in a single file so reset the DOM.
+    while (document.body.firstChild) {
+      document.body.removeChild(document.body.firstChild);
+    }
+    jest.clearAllMocks();
+  }
+
+  beforeEach(() => {
+    mockHeadless();
+  });
+
+  afterEach(() => {
+    cleanup();
+  });
+
+  describe('when the result is of type Salesforce', () => {
+    it('should open the result link in a Salesforce console subtab', async () => {
+      const element = createTestComponent({...mockSalesforceResult});
+      await flushPromises();
+
+      const linkSalesforce = element.shadowRoot.querySelector('a');
+      linkSalesforce.click();
+
+      const {pageReference} = getNavigateCalledWith();
+
+      expect(pageReference.attributes.recordId).toBe(
+        mockSalesforceResult.result.raw.sfid
+      );
+    });
+
+    describe('when the result is a knowledge article', () => {
+      it('should open the result link in a Salesforce console subtab', async () => {
+        const element = createTestComponent({...mockKnowledgeArticleResult});
+        await flushPromises();
+
+        const linkSalesforce = element.shadowRoot.querySelector('a');
+        linkSalesforce.click();
+
+        const {pageReference} = getNavigateCalledWith();
+
+        expect(pageReference.attributes.recordId).toBe(
+          mockKnowledgeArticleResult.result.raw.sfkavid
+        );
+      });
+    });
+  });
+  describe('when the result is NOT of type salesforce', () => {
+    it('should open the result link in the current browser tab', async () => {
+      const element = createTestComponent({...mockDefaultResult});
+      await flushPromises();
+
+      const link = element.shadowRoot.querySelector('a');
+      link.click();
```

Comment:
Why do you need to click it btw?
Your test only look at the attributes `href` and `target`.
Those are not impacted after you've clicked on it right?

Reply:
you are right, for the other 2 tests I need the click but not this one